token-lending/program/src/processor.rs

```diff
@@ -1884,6 +1911,46 @@ fn get_pyth_price(pyth_price_info: &AccountInfo, clock: &Clock) -> Result<Decima
     Ok(market_price)
 }
 
+fn get_switchboard_price(
+    switchboard_feed_info: &AccountInfo,
+    clock: &Clock,
+) -> Result<Decimal, ProgramError> {
+    const STALE_AFTER_SLOTS_ELAPSED: u64 = 30;
+
+    let account_buf = switchboard_feed_info.try_borrow_data()?;
+    if account_buf[0] != SwitchboardAccountType::TYPE_AGGREGATOR as u8 {
+        msg!("switchboard address not of type aggregator");
+        return Err(LendingError::InvalidAccountInput.into());
+    }
+
+    let aggregator: AggregatorState = get_aggregator(switchboard_feed_info)?;
+    // if aggregator.version != 1 {
+    //     msg!("switchboard version incorrect");
+    //     return Err(LendingError::InvalidAccountInput.into());
+    // }
+    let round_result: RoundResult = get_aggregator_result(&aggregator)?;
+
+    let slots_elapsed = clock
+        .slot
+        .checked_sub(round_result.round_open_slot.unwrap())
+        .ok_or(LendingError::MathOverflow)?;
+    if slots_elapsed >= STALE_AFTER_SLOTS_ELAPSED {
+        msg!("Switchboard oracle price is stale");
+        return Err(LendingError::InvalidOracleConfig.into());
+    }
+
+    let price_float = round_result.result.unwrap_or(0.0);
+
+    // we just do this so we can parse coins with low usd value
+    let price_quotient = 10u64.pow(9);
+
+    let price = ((price_quotient as f64) * price_float).round() as u128;
+
+    let decimal_price = Decimal::from(price).try_div(price_quotient)?;
+
+    Ok(decimal_price)
```

Comment:
Same thing here
`Decimal::from(price).try_div(price_quotient)`

Reply:
yeh i guess i was just being more explicit/ if add a 3rd thing but ill just do it that way sure

Reply:
Yeah i guess it doesn't really matter, but i guess its shorter or something.